js/main.js

```diff
@@ -7,51 +7,95 @@ var $photoUrl = document.querySelector('.photo-url');
 var $notes = document.querySelector('.notes');
 var $photoUpdate = document.querySelector('.photo');
 var $entriesTab = document.querySelector('a');
-var $p = document.querySelector('p');
 var $ul = document.querySelector('ul');
 var $newButton = document.querySelector('.new-button');
 var $views = document.querySelectorAll('.view');
+var $h1 = document.querySelector('h1');
+var $noEntries = document.querySelector('.no-entries');
 
 $photoUrl.addEventListener('input', function (event) {
   $photoUpdate.setAttribute('src', event.target.value);
 });
 
 $form.addEventListener('submit', function (event) {
   event.preventDefault();
-  var submission = {
-    title: $title.value,
-    photoUrl: $photoUrl.value,
-    notes: $notes.value,
-    entryId: data.nextEntryId
-  };
-  data.nextEntryId++;
-  data.entries.push(submission);
-  $ul.prepend(renderEntry(submission));
+  if (data.editing === null) {
+    var submission = {
+      title: $title.value,
+      photoUrl: $photoUrl.value,
+      notes: $notes.value,
+      entryId: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.push(submission);
+    $ul.prepend(renderEntry(submission));
+  } else {
+    var editedSubmission = {
+      title: $title.value,
+      photoUrl: $photoUrl.value,
+      notes: $notes.value,
+      entryId: data.editing
+    };
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.editing === data.entries[i].entryId) {
+        data.entries[i] = editedSubmission;
+        break;
+      }
+    }
+    var $lis = document.querySelectorAll('li');
+    var editDomTree = renderEntry(editedSubmission);
+    for (i = 0; i < $lis.length; i++) {
+      if (parseInt($lis[i].getAttribute('data-entry-id')) === data.editing) {
+        $lis[i].replaceWith(editDomTree);
+      }
+    }
+  }
   $photoUpdate.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $form.reset();
   changeView('entries');
-  if (data.entries.length > 0) {
-    $p.setAttribute('class', 'hidden');
-  }
+  location.reload();
```

Comment:
WOAHHHHHH

You should **_never_** use `location.reload`. This reloads the entire page, which is the most intensive operation you can do with a web app. Why is this necessary?